specification/cognitiveservices/data-plane/CustomDecisionService/v2.0/CustomDecisionService.json

```diff
@@ -0,0 +1,358 @@
+          },
+          {
+            "type": "string",
+            "name": "eventId",
+            "in": "path",
+            "required": true,
+            "x-nullable": false,
+            "description": "The event id this reward should bne joined with."
+          },
+          {
+            "name": "reward",
+            "in": "body",
+            "schema": {},
+            "x-nullable": true,
+            "description": "The reward should be a float, unless the application is configured to use a custom reward function. If a custom reward function is configured, the scheme of the reward is dependent on the function."
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": ""
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DecisionRequestCollection": {
+      "type": "object",
+      "description": "A batch of decision requests. They have to target the same app for now.",
+      "additionalProperties": false,
+      "properties": {
+        "decisions": {
+          "type": "array",
+          "description": "A batch of decision requests. They have to target the same app for now.",
+          "items": {
+            "$ref": "#/definitions/DecisionRequest"
+          }
+        }
+      }
+    },
+    "DecisionRequest": {
+      "type": "object",
+      "description": "A Decision request",
+      "additionalProperties": false,
+      "properties": {
+        "site": {
+          "type": "string",
+          "description": "For future use"
+        },
+        "shared": {
+          "description": "The shared set of features (between actions). E.g. user demographics.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/DecisionFeatures"
+            }
+          ]
+        },
+        "actions": {
+          "type": "array",
+          "description": "The set of actions the Custom Decision Service should pick from",
+          "items": {
+            "$ref": "#/definitions/DecisionFeatures"
+          }
+        },
+        "actionSets": {
+          "type": "array",
+          "description": "The set of server-side action sets that should be resolved to actions.",
+          "items": {
+            "$ref": "#/definitions/DecisionActionSet"
+          }
+        }
+      }
+    },
+    "DecisionFeatures": {
+      "type": "object",
+      "description": "Specifies a set of ML features, either directly ",
+      "additionalProperties": false,
+      "required": [
+        "lock"
+      ],
+      "properties": {
+        "ids": {
+          "type": "array",
+          "description": "References features from a server-side feature set.",
+          "items": {
+            "$ref": "#/definitions/DecisionReferenceId"
+          }
+        },
+        "features": {
+          "type": "array",
+          "description": "Vowpal Wabbit style features",
+          "items": {
+            "type": "string"
+          },
+          "OverrideItemType": "string"
```

Comment:
What is this? This is non-standard Swagger and fails our validation, please remove 🙂 (note: 2 more occurrences below)

Reply:
hmm.. not sure how the OverrideItemTypes ended up there, but features need to be of type object in the generate client.

In general this should be object as this part accepts arbitrary JSON.

Reply:
why is it modeled as an array then? 😄 all I can say is, this is non-standard Swagger and rejected by our validation tools - the code generator itself is a little more flexible, but generates features as `string[]` (well, in C# at least) since that's how it's specified right now 😉